tilavarauspalvelu/integrations/keyless_entry/client.py

```diff
@@ -0,0 +1,257 @@
+
+    ##################
+    # Helper methods #
+    ##################
+
+    @classmethod
+    def _parse_reservation_unit_response(cls, data: dict[str, Any]) -> PindoraReservationUnitResponse:
+        try:
+            return PindoraReservationUnitResponse(
+                reservation_unit_id=uuid.UUID(data["reservation_unit_id"]),
+                name=data["name"],
+                keypad_url=data["keypad_url"],
+            )
+
+        except KeyError as error:
+            msg = f"Missing key in reservation unit response from Pindora: {error}"
+            raise PindoraAPIError(msg) from error
+
+        except ValueError as error:
+            msg = f"Invalid value in reservation unit response from Pindora: {error}"
+            raise PindoraAPIError(msg) from error
+
+    @classmethod
+    def _parse_reservation_response(cls, data: dict[str, Any]) -> PindoraReservationResponse:
+        try:
+            return PindoraReservationResponse(
+                reservation_unit_id=uuid.UUID(data["reservation_unit_id"]),
+                access_code=data["access_code"],
+                access_code_keypad_url=data["access_code_keypad_url"],
+                access_code_phone_number=data["access_code_phone_number"],
+                access_code_sms_number=data["access_code_sms_number"],
+                access_code_sms_message=data["access_code_sms_message"],
+                access_code_valid_minutes_before=data["access_code_valid_minutes_before"],
+                access_code_valid_minutes_after=data["access_code_valid_minutes_after"],
+                access_code_generated_at=datetime.datetime.fromisoformat(data["access_code_generated_at"]),
+                begin=datetime.datetime.fromisoformat(data["begin"]),
+                end=datetime.datetime.fromisoformat(data["end"]),
+            )
+
+        except KeyError as error:
+            msg = f"Missing key in reservation response from Pindora: {error}"
+            raise PindoraAPIError(msg) from error
+
+        except ValueError as error:
+            msg = f"Invalid value in reservation response from Pindora: {error}"
+            raise PindoraAPIError(msg) from error
+
+    @classmethod
+    def _build_url(cls, endpoint: str) -> str:
+        if not settings.PINDORA_API_URL:
+            msg = "'PINDORA_API_URL' environment variable must to be configured."
+            raise PindoraAPIConfigurationError(msg)
+
+        base_url = settings.PINDORA_API_URL.removesuffix("/")
+        return f"{base_url}/{endpoint}"
+
+    @classmethod
+    def _get_headers(cls, headers: dict[str, Any] | None) -> dict[str, str]:
+        if not settings.PINDORA_API_KEY:
+            msg = "'PINDORA_API_KEY' environment variable must to be configured."
+            raise PindoraAPIConfigurationError(msg)
+
+        return {
+            "Pindora-Api-Key": str(settings.PINDORA_API_KEY),
+            "Accept": "application/vdn.varaamo-pindora.v1+json",
```

Comment:
Just `application/json` in the OpenAPI description, after all.

Reply:
Asked about this in Keyless Entry Trello, we can switch it if they have decided in some other versioning method.